### PR TITLE
Make sure data examples achieve a single task

### LINKED ISSUE #5888: Replace the title of the "Downloading and plotting an HMI magnetogram" example with something more clear
### Provide a general description of the issue or problem.

The "Downloading and plotting an HMI magnetogram" example offers an explanation of why an HMI map appears to be "upside down" when you plot it. However, there are still semi-frequent issues or mailing list posts asking this question. I think we should rename this example something far more explicit ("Why is my HMI magnetogram upside down?") that makes it immediately obvious to users that that is the example they should look at. The current title does not do that.

### PR DESCRIPTION
**Pull request by dstansby**
As part of the documentation reorganisation we will have the four types of documentation described at https://diataxis.fr/. One of these is "How-to guides", which are 
> recipes, directions that guide the reader through the steps to achieve a specific end.

We already have these guides to some extent in our example gallery.

This PR is the first part of work to make sure the files in our example gallery achieve **one** specific task. I don't think any of the delted content in this PR needs to be moved elsewhere as it's covered by other examples, but in PRs for the other example categories we may want to move some stuff to entries in a Tutorials section.

Fixes #5888

### CODE REVIEW

File: examples/acquiring_data/downloading_lascoC2.py
Comment (nabobalis) [inline examples/acquiring_data/downloading_lascoC2.py:1]: 👍 

I think back in the day we had this due to the metadata needing to be manually corrected.

File: examples/acquiring_data/goes_xrs_example.py
Comment (nabobalis) [inline examples/acquiring_data/goes_xrs_example.py:78]: I wonder if this needs to stay but this becomes a tutorial or something? Or maybe text in the GOES timeseries class about the quality flags?

File: examples/map_transformations/downloading_hmi.py
Comment (nabobalis) [inline examples/map_transformations/downloading_hmi.py:3]:
```suggestion
Solar physicists hate it, One Simple Trick to rotate your upside-down HMI image
```
I think this needs to be more Buzzfeed style

File: examples/acquiring_data/downloading_cutouts.py
Comment (nabobalis) [inline examples/acquiring_data/downloading_cutouts.py:112]: We lose one of our only gallery animations tho and I am sad to see that go.

File: examples/map_transformations/downloading_hmi.py
Comment (dstansby) [reply examples/map_transformations/downloading_hmi.py:3]: > They opened a HMI FITS file and you won't believe what happened next

File: examples/acquiring_data/downloading_cutouts.py
Comment (dstansby) [reply examples/acquiring_data/downloading_cutouts.py:92]: Is this hot take just to make the example pass? If so I'm 👍 (and I can rename the example to "searching for cutouts")

File: examples/acquiring_data/goes_xrs_example.py
Comment (dstansby) [reply examples/acquiring_data/goes_xrs_example.py:78]: 👍 I can move this to a temporary folder for future tutorial material so it's not lost

File: examples/acquiring_data/downloading_cutouts.py
Comment (dstansby) [reply examples/acquiring_data/downloading_cutouts.py:112]: Other suggestions welcome - "Plotting an animation" seems like it's a single task that only needs one example though 😢  - we can fill up the future tutorials with them though!

Comment (nabobalis) [reply examples/acquiring_data/downloading_cutouts.py:112]: Yeah, it should go, we should check if we have an animation sequence example somewhere and not if, a new PR/issue to add one in future.

Comment (dstansby) [reply examples/acquiring_data/downloading_cutouts.py:112]: https://docs.sunpy.org/en/stable/generated/gallery/plotting/arrayanimatorwcs.html#sphx-glr-generated-gallery-plotting-arrayanimatorwcs-py already exsists 👍

**Comment by dstansby** [comment]
Sorry for the reviewer pain, I've updated my commit and force pushed. All your comments have been addressed - might be worth having a particular look at the cutout example, which I've modified quite a bit. Tested and working locally for me.

**Comment by nabobalis** [comment]
I think a tracking issue for the tutorials folder should be opened once this is merged so we don't forget.

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Needs a second review but LGTM

File: examples/acquiring_data/downloading_lascoC2.py
Comment (Cadair) [inline examples/acquiring_data/downloading_lascoC2.py:3]: I get that this isn't doing anything particularly "interesting" but if someone wants to get a C2 image they are going to find this and follow it. What's the harm in keeping it?

File: examples/acquiring_data/downloading_cutouts.py
Comment (Cadair) [inline examples/acquiring_data/downloading_cutouts.py:100]: Why not also show download and load into a Map? It's all part of a single "task" in my opinion, you aren't going to want to search for and download stuff without wanting to load them after?

File: examples/acquiring_data/downloading_cutouts.py
Comment (Cadair) [inline examples/acquiring_data/downloading_cutouts.py:114]: The only reason I am sad about removing this is that this was a nice little gif on the gallery page, and much more eye catching than the now complete lack of a preview image.

**Review by Cadair — COMMENTED** [review]
While I get the desire to be pure to the methodology, I do think we also need to consider the experience of someone scrolling through the gallery, if you make a plot after you search for / download data then you get a nice preview image where someone can say "hey that looks like what I am after" if we always just stop at `Fido.fetch` then you don't get that.

This is particularly relevant to the JSOC cutout example here.

**Comment by dstansby** [comment]
> if you make a plot after you search for / download data then you get a nice preview image where someone can say "hey that looks like what I am after" if we always just stop at `Fido.fetch` then you don't get that.

👍 good point - I will go through these and put back in or add new simple plots wherever we download data.

File: examples/acquiring_data/downloading_cutouts.py
Comment (wtbarnes) [reply examples/acquiring_data/downloading_cutouts.py]: FWIW, this code can now be greatly simplified to just set `norm=...` in `sequence.plot()`.

Comment (dstansby) [reply examples/acquiring_data/downloading_cutouts.py]: Is that a blocking request for this PR? I'd advocate for merge this as it's an improvement and this plotting code isn't changed, and we can sort of all the norms (if there are more?) in another PR.

**Comment by dstansby** [comment]
No, I don't think we should either! This PR now doesn't remove any final images.

File: examples/acquiring_data/downloading_cutouts.py
Comment (wtbarnes) [reply examples/acquiring_data/downloading_cutouts.py]: Nope not blocking. I should've cleaned these up a long time ago.

File: examples/map_transformations/downloading_hmi.py
Comment (wtbarnes) [inline examples/map_transformations/downloading_hmi.py:3]: 👍

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Doc failure on: 
"/home/runner/work/sunpy/sunpy/docs/guide/data_types/maps.rst:26: WARNING: undefined label: 'sphx_glr_generated_gallery_acquiring_data_2011_06_07_sampledata_overview.py'"

**Comment by Cadair** [comment]
This should have had two approvals.

File: examples/map_transformations/downloading_hmi.py
Comment (Cadair) [inline examples/map_transformations/downloading_hmi.py:7]:
```suggestion
it appears with solar North pointing up.
```

File: examples/acquiring_data/downloading_cutouts.py
Comment (Cadair) [inline examples/acquiring_data/downloading_cutouts.py:37]:
```suggestion
# Next, we will use the coordinate frame from this map to definte the top right and bottom
```

File: examples/acquiring_data/downloading_cutouts.py
Comment (Cadair) [inline examples/acquiring_data/downloading_cutouts.py:38]:
```suggestion
# left coordinates we want for the cutout request.
```

**Review by Cadair — CHANGES_REQUESTED** [review]

**Comment by nabobalis** [comment]
> This should have had two approvals.

The original version for sure.

**Comment by Cadair** [comment]
The original version of what?

**Comment by nabobalis** [comment]
> The original version of what?

Of this PR

**Comment by Cadair** [comment]
There was another version of this PR? I am very confused.

**Comment by nabobalis** [comment]
It's ok.

**Pull request merged**
